### PR TITLE
Issue 56: motion detail

### LINKED ISSUE #56: Add motion detail in API proxy

### PR DESCRIPTION
**Pull request by jdomingos**
Add `councilMotionDetail` query plus some refactoring for the motion detail export, affecting `councilMotions`.

UPDATE: it works!

Closes #56

### CODE REVIEW

**Comment by jdomingos** [comment]
@franjs @ishrak-imam feel free to review, since I've confirmed it works! Thanks 😄

File: subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts [hunk 1]
```diff
@@ -23,21 +25,32 @@ export async function councilMotions(
...
-        ...getCallParams(motion.proposal),
-      };
-
-      return {
-        hash: String(motion.hash),
-        proposal,
-        votes: motion.votes ? await getVotes(motion.votes, accountService, api) : undefined,
-        votingStatus: motion.votes ? getVotingStatus(motion.votes, councilMembers.length, bestNumber, api) : undefined,
-      };
+      return (await getMotionDetails(motion, accountService, api, councilMembers, bestNumber)) as CouncilMotion;
```
Comment (franjs) [inline subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts:28]: There is no need to async/await here, you are already doing `Promise.all`. Why do you need to do `as CouncilMotion`?

File: subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts [hunk 2]
```diff
@@ -23,21 +25,32 @@ export async function councilMotions(
...
+  ]);
+
+  if (!motion) {
+    return null;
+  }
+
+  const accountService = new AccountsService(api);
+  const councilMembers = electionsInfo.members;
+
+  return await getMotionDetails(motion, accountService, api, councilMembers, bestNumber);
```
Comment (franjs) [inline subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts:51]: NIT: no need to `await` here

File: subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts [hunk 3]
```diff
@@ -107,3 +120,23 @@ function getVotingStatus(
...
   };
 }
+
+async function getMotionDetails(
+  motion: DeriveCollectiveProposal,
+  accountService: AccountsService,
+  api: Context['api'],
+  councilMembers: [AccountId, Balance][],
+  bestNumber: BlockNumber,
+): Promise<CouncilMotion | null> {
```
Comment (franjs) [inline subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts:130]: Why `| null`?

Comment (jdomingos) [reply subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts:130]: Oh you're right, I refactored but decided to return null on the function above 👍  Will change

File: subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts (hunk 1 above)
Comment (jdomingos) [reply subschemas/substrate-chain/src/resolvers/Query/councilMotions.ts:28]: Yeah, right, it's not ok - I'll change it - it was the result of a refactoring that I didn't cleanup properly 👍

**Review by franjs — APPROVED** [review]

**Pull request merged**